src/utils/shared.ts

```diff
@@ -8,10 +8,19 @@ export const mapToObject = (inputMap: Map<string, unknown>): object => {
   return obj;
 };
 
-export const clearSecretEmptyValue = (obj: any) => {
-  return Object.keys(obj).map((key) => {
-    if (_.isEmpty((obj as { [key: string]: any })[key])) {
-      (obj as { [key: string]: any })[key] = '';
+/**
+ * Returns a new object that does not contain empty objects as values
+ * and leaves the passed in value untouched.
+ * @param obj
+ * @returns
+ */
+export const clearEmptyObjectValues = (obj: any): any => {
+  const answer: any = { ...obj };
```

Comment:
this may only do a shallow copy

Reply:
I think it's fine provided we're all considering these configuration objects as immutable.  It really shouldn't be necessary to modify these object's attributes in the views, and even configuring should be producing a new configuration object vs directly modifying the attributes of the existing one.